### DISCUSSION TITLE
entity_vector_length mismatching for the entity linker training

### DISCUSSION THREAD

**Discussion by manxing-du** [explosion/spaCy#11628] [Help: Coding & Implementations]
In the config file, the entity_vector_length is 64, but the vectors from the "en_core_web_lg" or "en_core_web_md" are with dimension 300. Could we find any pre-trained model which provides the 64 dimension entity embeddings ?

```
[components.entity_linker]
factory = "entity_linker"
entity_vector_length = 64
```

**Comment by polm** [comment]
Just a note - I moved this to Discussions because it looks more like a question than a bug report.

I don't think any of our pretrained models have vectors with size 64, but note that's just a placeholder value - the intent is that you change it to match the vectors you're actually using. It's not a recommendation or requirement that you use vectors with size 64.

**Reply by polm** [reply]
One thing about this that's a little confusing is that with the sample Emerson project, if you modify the `entity_vector_length` in the config it doesn't affect anything. That may be surprising, but it happens because in that project, the KB is built ahead of time, and simply loaded at training. When loading a serialized KB, it's not checked against the size listed in the config - instead the model is just updated to use the size of the serialized KB. 

Using a serialized KB is probably the standard way you'd do this, and in that case the config setting isn't going to be meaningful. But if you create an EntityLinker in code directly, for example, then the specified size will work as expected.
